### ISSUE TITLE
Unsupported project types are included in the processing if they are referenced from a supported project type.

### ISSUE THREAD

**Issue by petercullen68** [CycloneDX/cyclonedx-dotnet#896]
We have a .sln that contains a mixture of .csproj and .vcxproj.  The initial prcocessing in GetSolutionProjectReferencesAsync correctly removed invalid projects calling Utils.IsSupportedProjectType.

However the code just below this then loops through the valid projects and calls _projectFileService.RecursivelyGetProjectReferencesAsync to return a list of project references.  It does not filter these references using Utils.IsSupportedProjectType.  In my case this causes the program to crash attempting to process a .vcxproj.

**Comment by jwfx** [comment]
What error messages are you getting?

**Comment by mtsfoni** [comment]
Should work till version 2.10

I will have a look soon, shouldn't be a big issue

**Comment by petercullen68** [comment]
@mtsfoni Thank you.  My quick fix to test was this (in RecursivelyGetProjectReferencesAsync)


```c#
            while (files.Count > 0)
            {
                var currentFile = files.Dequeue();

                if (!Utils.IsSupportedProjectType(currentFile))
                {
                    continue;
                }
```

**Comment by petercullen68** [comment]
@jwfx The app attempts to process a C++ nuget package from the .vcxproj and throws an error trying to process it.

**Comment by petercullen68** [comment]
I have opened a pull request to fix this.  It has been tested and works in my case.

**Comment by mtsfoni** [comment]
Merged it. Released in 4.1.0
